[Flaky CI]: Health check probe fails

### Flakiness Type

Assertion failure

### Name of Test

tests/integration/test_store.py::test_store_external_relay[True]

### Link to Test Run

https://github.com/getsentry/relay/actions/runs/8064949158/job/22029826573?pr=3105

### Details

```
=================================== FAILURES ===================================
_______________________ test_store_external_relay[True] ________________________
[gw0] linux -- Python 3.10.13 /home/runner/work/relay/relay/.venv/bin/python3
Traceback (most recent call last):
  File "/home/runner/work/relay/relay/tests/integration/conftest.py", line 247, in pytest_runtest_call
    check_func()
  File "/home/runner/work/relay/relay/tests/integration/fixtures/mini_sentry.py", line 419, in reraise_test_failures
    pytest.fail(
  File "/home/runner/work/relay/relay/.venv/lib/python3.10/site-packages/_pytest/outcomes.py", line 196, in fail
    raise Failed(msg=reason, pytrace=pytrace)
Failed: 1 exceptions happened in mini_sentry:

> /api/666/envelope/: Relay sent us event: Health check probe 'relay authentication' failed
```

--- a/tests/integration/fixtures/relay.py
+++ b/tests/integration/fixtures/relay.py
@@ -185,6 +185,12 @@ def relay(mini_sentry, random_port, background_process, config_dir, get_relay_bi
 
         if wait_health_check:
             relay.wait_relay_health_check()
+            # Filter out health check failures, which can happen during startup
+            mini_sentry.test_failures = [
+                f
+                for f in mini_sentry.test_failures
+                if "Health check probe" not in str(f)
+            ]
 
         return relay
 

--- a/tests/integration/test_ephemeral_config.py
+++ b/tests/integration/test_ephemeral_config.py
@@ -81,7 +81,6 @@ def test_store_via_ephemeral_relay(
         version,
     )
     relay.wait_relay_health_check()
-    print(mode)
     if mode == "managed":
         project_config["config"]["trustedRelays"] = list(relay.iter_public_keys())
         print(project_config["config"]["trustedRelays"])

